packages/store/src/utils/angular.ts

```diff
@@ -0,0 +1,18 @@
+import { getPlatform, COMPILER_OPTIONS, CompilerOptions, PlatformRef } from '@angular/core';
+import { memoize } from './memoize';
+
+function _isAngularInTestMode() {
+  const platformRef: PlatformRef | null = getPlatform();
+  if (!platformRef) return false;
+  const compilerOptions = platformRef.injector.get(COMPILER_OPTIONS, null);
+  if (!compilerOptions) return false;
+  const isInTestMode = compilerOptions.some((item: CompilerOptions) => {
+    const providers = (item && item.providers) || [];
```

Comment:
how hard is the operation? How many iterations will be in production mode?

the number of providers may be a huge?

Reply:
These compiler options will iterate over no more than 10 providers and this list does not get affected by the providers in your app. These are stock standard.